### PR TITLE
Add support for optimism granite hardfork

### PR DESCRIPTION
**Pull request by BrianBland**
Adds optimism granite hardfork definition and support for genesis file config, and defines the activation timestamp for op-sepolia and base-sepolia.
This PR depends on the following dependency changes:
* op-alloy: https://github.com/alloy-rs/op-alloy/pull/31
* revm: https://github.com/bluealloy/revm/pull/1685

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm, pending revm bump

**Review by Rjected — COMMENTED** [review]
do you mind adding a forkid test to `base_sepolia_forkids`?

File: crates/chainspec/src/spec.rs [hunk 1]
```diff
@@ -758,6 +758,8 @@ impl From<Genesis> for ChainSpec {
             (OptimismHardfork::Ecotone.boxed(), genesis_info.ecotone_time),
             #[cfg(feature = "optimism")]
             (OptimismHardfork::Fjord.boxed(), genesis_info.fjord_time),
+            #[cfg(feature = "optimism")]
+            (OptimismHardfork::Granite.boxed(), genesis_info.granite_time),
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:762]: this only adds it to base-sepolia?

we should also add it to op-sepolia?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/chainspec/src/spec.rs (hunk 1 above)
Comment (BrianBland) [reply crates/chainspec/src/spec.rs:762]: This change should apply Granite to any ChainSpec based on the genesis file config. The base-sepolia and op-sepolia activation timestamps are configured towards the end of `crates/ethereum-forks/src/hardfork/optimism.rs`

**Review by mattsse — APPROVED** [review]

**Pull request merged**